From 48b6a7b311272896be9212e170fcee8d1da25e79 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Thu, 2 Jun 2022 16:16:35 +0200
Subject: [PATCH] PHP: Fixed php_module_startup() call for PHP 8.2.

PHP 8.2 changed the prototype of the function, removing the last
parameter.

Signed-off-by: Remi Collet <remi@remirepo.net>
Cc: Timo Stark <t.stark@nginx.com>
Cc: George Peter Banyard <girgias@php.net>
Tested-by: Andy Postnikov <apostnikov@gmail.com>
Acked-by: Andy Postnikov <apostnikov@gmail.com>
Reviewed-by: Andrew Clayton <a.clayton@nginx.com>
Signed-off-by: Alejandro Colomar <alx@nginx.com>
---
 auto/modules/php   | 4 ++++
 docs/changes.xml   | 6 ++++++
 src/nxt_php_sapi.c | 4 ++++
 3 files changed, 14 insertions(+)

--- auto/modules/php.orig
+++ auto/modules/php
@@ -149,7 +149,11 @@ nxt_feature_test="
     #include <php_main.h>
 
     int main() {
+    #if (PHP_VERSION_ID < 80200)
         php_module_startup(NULL, NULL, 0);
+    #else
+        php_module_startup(NULL, NULL);
+    #endif
         return 0;
     }"
 
--- src/nxt_php_sapi.c.orig
+++ src/nxt_php_sapi.c
@@ -1150,7 +1150,11 @@ nxt_php_vcwd_chdir(nxt_unit_request_info_t *req, u_char *dir)
 static int
 nxt_php_startup(sapi_module_struct *sapi_module)
 {
+#if (PHP_VERSION_ID < 80200)
     return php_module_startup(sapi_module, &nxt_php_unit_module, 1);
+#else
+    return php_module_startup(sapi_module, &nxt_php_unit_module);
+#endif
 }
 
 
